features/vaadin-dashlets/dashlet-ksc/src/main/java/org/opennms/features/vaadin/dashboard/dashlets/KscDashlet.java

```diff
@@ -430,6 +423,13 @@ public Component getComponent() {
         });
     }
 
+    String determineNodeIdByResourceId(ResourceId resourceId){
+        OnmsResource parentResource = m_resourceDao.getResourceById(resourceId).getParent();
+        OnmsNode node = ResourceTypeUtils.getNodeFromResource(parentResource);
```

Comment:
We should re-use this node object instead of making subsequent NodeDao calls with the extract ID.